src/plugins/intel_gpu/src/graph/impls/ocl/non_max_suppression.cpp

```diff
@@ -101,21 +102,36 @@ struct non_max_suppression_impl : typed_primitive_impl_ocl<non_max_suppression>
                 params.soft_nms_sigma = get_value<float>(node);
             } else {
                 params.soft_nms_sigma_type = kernel_selector::NmsArgType::Input;
-                params.inputs.push_back(convert_data_tensor(node.get_output_layout()));
+                params.inputs.push_back(convert_data_tensor(impl_param->output_layout));
             }
         }
 
+        auto get_second_output_node_idx = [&] {
+            size_t offset = 2;
+            offset += arg.has_num_select_per_class();
+            offset += arg.has_iou_threshold();
+            offset += arg.has_score_threshold();
+            offset += arg.has_soft_nms_sigma();
+            return offset;
+        };
+
+        auto get_third_output_node_idx = [&] {
```

Comment:
These lambdas have pretty much the same code. Maybe single lambda with parameter would be better?
```cpp
auto get_output_node_idx = [&](size_t output_idx) {...}
```

Reply:
Fixed, thanks!